lib/segment/src/types.rs

```diff
@@ -1954,12 +1954,15 @@ pub struct WithPayload {
 #[serde(deny_unknown_fields, rename_all = "snake_case")]
 pub struct Filter {
     /// At least one of those conditions should match
+    #[serde(default, deserialize_with = "deserialize_filter_item")]
     #[validate]
     pub should: Option<Vec<Condition>>,
     /// All conditions must match
+    #[serde(default, deserialize_with = "deserialize_filter_item")]
     #[validate]
     pub must: Option<Vec<Condition>>,
     /// All conditions must NOT match
+    #[serde(default, deserialize_with = "deserialize_filter_item")]
     #[validate]
     pub must_not: Option<Vec<Condition>>,
 }
```

Comment:
This works effectively, but it's hard to generate an openapi schema from this

Reply:
@coszio thanks for the review. although using untagged enums might be a clean way of fixing this, it requires a lot of changes in different parts of the code. so how about using `#[schemars(schema_with = "...")]` in order to generate the desired openapi schema for this?

Reply:
Sure, let's see how that looks. In my opinion it looks like this parallel implementation might be harder to maintain. 

On another note, how about making the function generic and renaming to `deserialize_one_or_many`